File: app/src/main/java/me/zhanghai/android/files/provider/common/PosixGroup.java
/*
 * Copyright (c) 2018 Hai Zhang <dreaming.in.code.zh@gmail.com>
 * All Rights Reserved.
 */

package me.zhanghai.android.files.provider.common;

import android.os.Parcel;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import java8.nio.file.attribute.GroupPrincipal;

public class PosixGroup extends PosixPrincipal implements GroupPrincipal {

    public PosixGroup(int id, @Nullable ByteString name) {
        super(id, name);
    }


    public static final Creator<PosixGroup> CREATOR = new Creator<PosixGroup>() {
        @NonNull
        @Override
        public PosixGroup createFromParcel(@NonNull Parcel source) {
            return new PosixGroup(source);
        }
        @NonNull
        @Override
        public PosixGroup[] newArray(int size) {
            return new PosixGroup[size];
        }
    };

    protected PosixGroup(@NonNull Parcel in) {
        super(in);
    }

    @Override
    public void writeToParcel(@NonNull Parcel dest, int flags) {
        super.writeToParcel(dest, flags);
    }
}

File: app/src/main/java/me/zhanghai/android/files/provider/common/PosixGroup.kt
/*
 * Copyright (c) 2018 Hai Zhang <dreaming.in.code.zh@gmail.com>
 * All Rights Reserved.
 */

package me.zhanghai.android.files.provider.common

import android.os.Parcel
import android.os.Parcelable
import java8.nio.file.attribute.GroupPrincipal

class PosixGroup : PosixPrincipal, GroupPrincipal {
    constructor(id: Int, name: ByteString?) : super(id, name)

    private constructor(source: Parcel) : super(source)

    companion object {
        @JvmField
        val CREATOR = object : Parcelable.Creator<PosixGroup> {
            override fun createFromParcel(source: Parcel): PosixGroup = PosixGroup(source)

            override fun newArray(size: Int): Array<PosixGroup?> = arrayOfNulls(size)
        }
    }
}
